### ISSUE TITLE
CKAN.ModuleNotFoundKraken Installing Kerbin Side Skyways error Transfer Window Planner

### ISSUE THREAD

**Issue by nobodyhasthis** [KSP-CKAN/NetKAN#3606]
Issue raised for unhandled exception but believe it is NetKan related due odd dependency. Sorry for any inconvenience caused if this not the case. Also getting CKAN-meta error message to which may require  new issue report details included below.
**CKAN Version / KSP:** 
CKAN 1.16.1-0-g2e91715
KSP 1.1.0 Pre-release build id = 01203  -  2016.04.12 at 22:50:28 CEST
**Operating System:** 
Windows 8.1
**The issue you are experiencing:** 
Any attempt to install Kerbin Side Skyways will produce an unhandled exception. With error message related to Transfer Window Planner. Which seems like an unrelated mod. This may be due to the way I have installed other kerbin side mods. Traditionally during a manual install all mods are included in "complete" all in one mod version. However no feedback apart from the error. I am left to guess at what the root cause.
**How to recreate this issue:** 
1. Use filter to search mods with "Kerbin Side" in name.
2. Install all of them except Ground Control and Skyways
3. Run Game
4. Install Kerbin Side Skyways to produce crash

optional step
5. Install Kerbin Side Ground Control which leads to file over write warning. Asking for another KSP-CKAN/CKAN-meta issue

**CKAN error codes (if applicable):** 
************** Exception Text **************
CKAN.ModuleNotFoundKraken: Cannot install TransferWindowPlanner, module not available
   at CKAN.CkanModule.FromIDandVersion(IRegistryQuerier registry, String mod, KSPVersion ksp_version)
   at CKAN.RelationshipResolver.<RelationshipResolver>c__AnonStorey0.<>m__0(String name)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at CKAN.RelationshipResolver..ctor(IEnumerable`1 module_names, RelationshipResolverOptions options, IRegistryQuerier registry, KSPVersion kspversion)
   at CKAN.MainModList.ComputeConflictsFromModList(IRegistryQuerier registry, IEnumerable`1 change_set, KSPVersion ksp_version)
   at CKAN.Main.<UpdateChangeSetAndConflicts>c__async1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CKAN.Main.<ModList_CellValueChanged>c__async0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c__DisplayClass2.<ThrowAsync>b__3(Object state)

**Second Over Error when selection of Kerbin Side Ground Control**
About to install...

 * Kerbin-Side Ground Control 0.3(cached)

Oh no! We tried to overwrite a file owned by another mod!
Please try a `ckan update` and try again.

If this problem re-occurs, then it maybe a packaging bug.
Please report it at:

https://github.com/KSP-CKAN/CKAN-meta/issues/new

Please including the following information in your report:

File           : GameData/KerbinSide/Flags/alphastrikelogo.png
Installing Mod : KerbinSideGroundControl 0.3
Owning Mod     : KerbinSide
CKAN Version   : v1.16.1-0-g2e91715 (beta)

Your GameData has been returned to its original state.
Error!

**Comment by plague006** [comment]
The method used for packaging `KerbinSide` mods was changed. They used to be packaged in a way that allowed multiple discrete modules to be installed side-by-side by CKAN however this is no longer the case.

With that said to resolve your situation you'll want to uninstall all KerbinSide modules and then re-install only 1 of them. In your case you want multiple KerbinSide mods so I suggest `Kerbin-Side Complete` which contains the full KerbinSide suite (but potentially additional KerbinSide mods that you don't want).

Additionally, `TransferWindowPlanner` is not 1.1.0-compatible.

**Comment by nobodyhasthis** [comment]
TL;DR I know how to take steps to totally avoid the unhandled exception. It still may be worth reporting just in case there is a bigger problem. Otherwise more that happy to close this one if this not adding any project value.

----

Normally it would just be a case of installing KerbinSide Complete which has everything. However it is still giving an option to still allowed multiple discrete modules. I was hoping that this could be made mutually exclusive. So either it is a sub module or the whole thing but not both. With a pop to tell the user to stop being silly. I have seen similar checks being done when say trying to install two mutually exclusive Tac LS configs. It's one or the other not both. Right now it giving us a chance to break things.

Also the TransferWindowPlanner error that being triggered with this has got me confused.
1. it is marked as 1.1 compatible- As in available for selection and does install. Should it be max version 1.0.5 to stop it appearing?

2. Why would trying to install KerbinSide help to throw an exception related to TransferWindowPlanner?

**Comment by politas** [comment]
The new scheme does indeed make all KerbinSide packages exclusive of all others, so once you have one installed, you won't be able to install any others, and CKAN should make that clear.

Not sure what's happening with TransferWindowPlanner, but assuming you have it selected, try going to the contents tab and re-downloading it to the cache.

**Comment by plague006** [comment]
TransferWindowPlanner shouldn't have a KSP compatibility higher than 1.0.5 (per https://github.com/KSP-CKAN/CKAN-meta/blob/master/TransferWindowPlanner/TransferWindowPlanner-v1.4.0.0.ckan). Further, in testing I'm unable to install TWP under KSP version 1.1.0.

**Comment by politas** [comment]
The larger issue of the reporting of a different mod to whatever's being installed is interesting.
I've seen a few reports recently that show that.

**Comment by plague006** [comment]
@nobodyhasthis would you mind re-creating the issue in https://github.com/KSP-CKAN/CKAN so that the coders/programmers have eyes on this?

**Comment by nobodyhasthis** [comment]
@plague006 More that happy to raise a new issue in main repository. Then only trouble is I do not think this error can now be recreated. At the time I must have refreshed available mods just before all of the metadata changes had gone through. Leaving CKAN in a temporary unstable state. Since initial report. 

1. TWP has dropped off latest list of available mods as predicted.  
2. The mutually exclusive check missing on Kerbin Side mods has started working. Blocking the user from selecting two conflicting mods.  

I am not sure what value a new issue would bring if the problem cannot be recreated. Do you thing I should copy the above error messages over to main repository?

Also. Now we have confirmed that the NetKAN is correct and these mods are appearing and installing correctly. I think that we could close this issue report. What do you think would be the best action?

**Comment by nobodyhasthis** [comment]
@plague006 I have seen this appearing again in another separate CKAN issue but without all the information. Same odd behaviour so have copied this issue across as suggested.

We should properly remove support label as close here.

**Comment by plague006** [comment]
Yep, closing this in favor of https://github.com/KSP-CKAN/CKAN/issues/1686. Thanks for moving it to where the coders will have eyes on it.
